Bonsai.Tinkerforge/BrickletDeviceNameConverter.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using Bonsai.Expressions;
+using Tinkerforge;
+using System.Threading;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace Bonsai.Tinkerforge
+{
+    internal class BrickletDeviceNameConverter : TypeConverter
+    {
+        public Dictionary<string, TinkerforgeHelpers.DeviceData> devices;
+        //private Stopwatch stopwatch;
+
+        public override bool GetStandardValuesSupported(ITypeDescriptorContext context)
+        {
+            return true;
+        }
+
+        public override bool GetStandardValuesExclusive(ITypeDescriptorContext context)
+        {
+            return true;
+        }
+
+        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+        {
+            return sourceType == typeof(string) || base.CanConvertFrom(context, sourceType);
+        }
+
+        // If we are doing the type conversion in a drop down, we are converting between a data object (DeviceData) and 
+        // a representative string in the drop down. We need to define the to/from conversion so that we can have a human
+        // readable representation of the device in the drop down, and use that string to look up the correct
+        // DeviceData on selection in the editor
+        public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
+        {
+            var casted = value as string;
+            return casted != null
+                ? devices[casted]
+                : base.ConvertFrom(context, culture, value);
+        }
+
+        public override object ConvertTo(ITypeDescriptorContext context, CultureInfo culture, object value, Type destinationType)
+        {
+            var casted = value as TinkerforgeHelpers.DeviceData;
```

Comment:
We could try to initialize the lookup here so we can convert UIDs to the full device data and return a formatted string starting with the numeric UID that we can then easily parse back in `ConvertFrom`.